Lib/test/test_tokenize.py

```diff
@@ -1873,6 +1878,33 @@ def test_indentation_semantics_retained(self):
         self.check_roundtrip(code)
 
 
+class InvalidPythonTests(TestCase):
+    def test_number_followed_by_name(self):
+        # See issue #gh-105549
+        source = "xsin(x)"
+        expected_tokens = [
+            TokenInfo(type=token.NAME, string='xsin', start=(1, 0), end=(1, 4), line='xsin(x)\n'),
+            TokenInfo(type=token.OP, string='(', start=(1, 4), end=(1, 5), line='xsin(x)\n'),
+            TokenInfo(type=token.NAME, string='x', start=(1, 5), end=(1, 6), line='xsin(x)\n'),
+            TokenInfo(type=token.OP, string=')', start=(1, 6), end=(1, 7), line='xsin(x)\n'),
+            TokenInfo(type=token.NEWLINE, string='', start=(1, 7), end=(1, 8), line='xsin(x)\n'),
+            TokenInfo(type=token.ENDMARKER, string='', start=(2, 0), end=(2, 0), line='')
+        ]
+
+        tokens = list(generate_tokens(StringIO(source).readline))
+        self.assertEqual(tokens, expected_tokens)
+
+    def test_number_starting_with_zero(self):
+        source = "01234"
+        expected_tokens = [
+            TokenInfo(type=token.NUMBER, string='01234', start=(1, 0), end=(1, 5), line='01234\n'),
```

Comment:
Why does `line` changes to append `\n` always at the end?

Reply:
This is getting fixed in https://github.com/python/cpython/issues/105564